### ISSUE TITLE
[Bug] Global archivelogs: true does not work for templates without outputs

### ISSUE THREAD

**Issue by Ark-kun** [argoproj/argo#3069]
Checklist:

 * [x] I've included the version.
 * [x] I've included reproduction steps.
 * [x] I've included the workflow YAML.
 * [ ] I've included the logs.
 
**What happened**:

When `archivelogs: true` is specified in `workflow-controller-configmap`, the logs are usually archived as an artifact. However this does not work for templates that have no outputs.

**What you expected to happen**:

I expect the logs to be always archived when `archivelogs: true` is specified in `workflow-controller-configmap`.

**How to reproduce it (as minimally and precisely as possible)**:

1. Set `archivelogs: true` is specified in `workflow-controller-configmap`
2. Run Hello world: https://github.com/argoproj/argo/blob/55ae0cb242a9cf6b390822ca6c0aa0868f5b06e3/examples/hello-world.yaml
3. Check whether any artifacts got uploaded

**Anything else we need to know?**:

When the template has outputs, the resolved Argo template annotation has `archiveLocation` which includes `archiveLogs`. However when template has no outputs, `archiveLocation` is completely missing from `workflows.argoproj.io/template`.

`workflows.argoproj.io/template`: 
```
"archiveLocation":{"archiveLogs":true,"s3":{"endpoint":"minio-service.default:9000","bucket":"mlpipeline","insecure":true,"accessKeySecret":{"name":"mlpipeline-minio-artifact","key":"accesskey"},"secretKeySecret":{"name":"mlpipeline-minio-artifact","key":"secretkey"},"key":"artifacts/file-passing-pipelines-w4cpp/file-passing-pipelines-w4cpp-758086958"}}
```

**Environment**:

- Argo version:
```
2.7.5
```

**Comment by alexec** [comment]
@dtaniwaki would you like to own.

**Comment by sarabala1979** [comment]
I am not able to reproduce this with the master branch. 

Workflow:
```
apiVersion: argoproj.io/v1alpha1
kind: Workflow
metadata:
  generateName: hello-world-
spec:
  entrypoint: whalesay
  templates:
  - name: whalesay
    container:
      image: docker/whalesay:latest
```
Configmap:
```
data:
  artifactRepository: |
    archiveLogs: true
    s3:
      bucket: my-bucket
      endpoint: minio:9000
      insecure: true
      accessKeySecret:
        name: my-minio-cred
        key: accesskey
      secretKeySecret:
        name: my-minio-cred
        key: secretkey
  containerRuntimeExecutor: pns
```
Pod Annotation:
```
 workflows.argoproj.io/outputs: '{"artifacts":[{"name":"main-logs","archiveLogs":true,"s3":{"endpoint":"minio:9000","bucket":"my-bucket","insecure":true,"accessKeySecret":{"name":"my-minio-cred","key":"accesskey"},"secretKeySecret":{"name":"my-minio-cred","key":"secretkey"},"key":"hello-world-szcpg/hello-world-szcpg/main.log"}}],"exitCode":"0"}'
    workflows.argoproj.io/template: '{"name":"whalesay","arguments":{},"inputs":{},"outputs":{},"metadata":{},"container":{"name":"","image":"docker/whalesay:latest","resources":{}},"archiveLocation":{"archiveLogs":true,"s3":{"endpoint":"minio:9000","bucket":"my-bucket","insecure":true,"accessKeySecret":{"name":"my-minio-cred","key":"accesskey"},"secretKeySecret":{"name":"my-minio-cred","key":"secretkey"},"key":"hello-world-szcpg/hello-world-szcpg"}}}'
  creationTimestamp: "2020-05-22T04:23:33Z"
```
Wait container log:
```
 k logs hello-world-szcpg  wait
time="2020-05-22T04:23:34Z" level=info msg="Starting Workflow Executor" version=latest+60d5fdc.dirty
time="2020-05-22T04:23:34Z" level=info msg="Creating PNS executor (namespace: argo, pod: hello-world-szcpg, pid: 7, hasOutputs: false)"
time="2020-05-22T04:23:34Z" level=info msg="Executor (version: latest+60d5fdc.dirty, build_date: 2020-05-11T16:39:00Z) initialized (pod: argo/hello-world-szcpg) with template:\n{\"name\":\"whalesay\",\"arguments\":{},\"inputs\":{},\"outputs\":{},\"metadata\":{},\"container\":{\"name\":\"\",\"image\":\"docker/whalesay:latest\",\"resources\":{}},\"archiveLocation\":{\"archiveLogs\":true,\"s3\":{\"endpoint\":\"minio:9000\",\"bucket\":\"my-bucket\",\"insecure\":true,\"accessKeySecret\":{\"name\":\"my-minio-cred\",\"key\":\"accesskey\"},\"secretKeySecret\":{\"name\":\"my-minio-cred\",\"key\":\"secretkey\"},\"key\":\"hello-world-szcpg/hello-world-szcpg\"}}}"
time="2020-05-22T04:23:34Z" level=info msg="Waiting on main container"
time="2020-05-22T04:23:37Z" level=info msg="main container started with container ID: bd19c15eb2e6c7a6601390921d3c46e69027087d60f6cd696a799bf635833017"
time="2020-05-22T04:23:37Z" level=info msg="Starting annotations monitor"
time="2020-05-22T04:23:37Z" level=info msg="Starting deadline monitor"
time="2020-05-22T04:23:37Z" level=warning msg="Ignoring wait failure: Failed to determine pid for containerID bd19c15eb2e6c7a6601390921d3c46e69027087d60f6cd696a799bf635833017: container may have exited too quickly. Process assumed to have completed"
time="2020-05-22T04:23:37Z" level=info msg="Main container completed"
time="2020-05-22T04:23:37Z" level=info msg="Saving logs"
time="2020-05-22T04:23:37Z" level=info msg="Annotations monitor stopped"
time="2020-05-22T04:23:37Z" level=info msg="S3 Save path: /tmp/argo/outputs/logs/main.log, key: hello-world-szcpg/hello-world-szcpg/main.log"
time="2020-05-22T04:23:37Z" level=info msg="Creating minio client minio:9000 using static credentials"
time="2020-05-22T04:23:37Z" level=info msg="Saving from /tmp/argo/outputs/logs/main.log to s3 (endpoint: minio:9000, bucket: my-bucket, key: hello-world-szcpg/hello-world-szcpg/main.log)"
time="2020-05-22T04:23:37Z" level=info msg="No output parameters"
time="2020-05-22T04:23:37Z" level=info msg="No output artifacts"
time="2020-05-22T04:23:37Z" level=info msg="No Script output reference in workflow. Capturing script output ignored"
time="2020-05-22T04:23:37Z" level=info msg="Capturing script exit code"
time="2020-05-22T04:23:37Z" level=info msg="Getting exit code of bd19c15eb2e6c7a6601390921d3c46e69027087d60f6cd696a799bf635833017"
time="2020-05-22T04:23:37Z" level=info msg="Annotating pod with output"
time="2020-05-22T04:23:37Z" level=info msg="Killing sidecars"
time="2020-05-22T04:23:37Z" level=info msg="Alloc=4767 TotalAlloc=11631 Sys=70592 NumGC=4 Goroutines=15"

```

Minio screen shot:
![image](https://user-images.githubusercontent.com/33908564/82631298-ff1b9f00-9ba9-11ea-8ac3-1fca931041e1.png)

**Comment by sarabala1979** [comment]
@Ark-kun

**Comment by Ark-kun** [comment]
Sorry for wasting your time. It turns out I got the versions wrong. We've upgraded Argo recently and I thought that our latest release already used Argo 2.7.5, but it actually still had Argo 2.3.0 (and since we use re-tagged images that was hard to spot).
I've verified that the issue is fixed in Argo 2.7.5.
Thank you.
